src/Avalonia.Controls/Converters/StringFormatConverter.cs

```diff
@@ -15,13 +13,25 @@ public class StringFormatConverter : IMultiValueConverter
 {
     public object? Convert(IList<object?> values, Type targetType, object? parameter, CultureInfo culture)
     {
-        try
-        {
-            return string.Format((string)values[0]!, values.Skip(1).ToArray());
-        }
-        catch (Exception e)
+        if (values != null &&
+            values.Count == 5 &&
+            values[0] is string format &&
+            values[1] is double &&
+            values[2] is double &&
+            values[3] is double &&
+            values[4] is double)
```

Comment:
Note: It would work correctly for progress bar. But it's less useful for other controls.

Reply:
This should just validate the first element aka `format`

Reply:
Not sure we are planning to use this converter for something else and currently we need to validate all values because during initialization Avalonia may pass "unset" as the value

Reply:
Note that this class is a part of the public API

Reply:
@kekekeks  Can we make it internal? I don't see any reason for it to be public